compiler/rustc_typeck/src/check/callee.rs

```diff
@@ -374,7 +374,25 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                     |p| format!("`{}` defined here returns `{}`", p, callee_ty),
                                 )
                             }
-                            _ => Some(format!("`{}` defined here", callee_ty)),
+                            _ => {
+                                match def {
+                                    // Emit a different diagnostic for local variables, as they are not
+                                    // type definitions themselves, but rather variables *of* that type.
+                                    Res::Local(id) => Some(format!(
+                                        "`{}` has type `{}`",
+                                        self.tcx.hir().name(id),
+                                        callee_ty
+                                    )),
+                                    Res::Def(kind, _) => {
+                                        if let Some(Namespace::ValueNS) = kind.ns() {
+                                            Some(format!("has type `{}`", callee_ty))
```

Comment:
This should show the name:

```suggestion
                                    Res::Def(kind, def_id) => {
                                        if let Some(Namespace::ValueNS) = kind.ns() {
                                            Some(format!("`{}` has type `{}`", self.tcx.def_path_str(def_id), callee_ty))
```